src/Extension/CoreExtension.php

```diff
@@ -1285,6 +1287,54 @@ public static function testEmpty($value): bool
         return '' === $value || false === $value || null === $value || [] === $value;
     }
 
+    /**
+     * Checks if a variable is a sequence.
+     *
+     * Traversables must be converted before being tested.
+     *
+     *    {# evaluates to true if the foo variable is a sequence #}
+     *    {% if foo is sequence %}
+     *        {# ... #}
+     *    {% endif %}
+     *
+     * @param mixed $value A variable
+     *
+     * @return bool true if the value is a sequence
+     *
+     * @internal
+     */
+    public static function testSequence($value)
+    {
+        if ($value instanceof \ArrayObject) {
+            $value = $value->getArrayCopy();
+        }
+        return \is_array($value) && array_is_list($value);
+    }
+
+    /**
+     * Checks if a variable is a mapping.
+     *
+     * Traversables must be converted before being tested.
+     *
+     *    {# evaluates to true if the foo variable is a mapping #}
+     *    {% if foo is mapping %}
+     *        {# ... #}
+     *    {% endif %}
+     *
+     * @param mixed $value A variable
+     *
+     * @return bool true if the value is a mapping
+     *
+     * @internal
+     */
+    public static function testMapping($value)
```

Comment:
```suggestion
    public static function testMapping($value): bool
```

Reply:
Thanks a lot. Done in https://github.com/twigphp/Twig/pull/3859/commits/798c8ffa10543aebe5e534fc043b5ba7c0580198